File: isAnagram/isAnagram.cs
public class Solution {
    public bool IsAnagram(string s, string t) {
        if (s.Length != t.Length) return false;

        Dictionary<char, int> dict = new Dictionary<char, int>();

        foreach (char c in s){
            if(dict.ContainsKey(c)) dict[c] += 1;
            else dict.Add(c, 1);
        }

        foreach (char c in t){
            if(dict.ContainsKey(c))
                if(dict[c] >= 1) dict[c] -= 1;
                else return false;
            else return false;
        }
        
        return true;
                
    }
}

File: isAnagram/isAnagram.py
class Solution:
    def isAnagram(self, s: str, t: str) -> bool:
        # ensure two words at the same length
        if(len(s) != len(t)): return False

        # create dictionary to store char occurence 
        table = {}
        for c in s:
            if c in table:
                table[c] = table[c] + 1
            else:
                table[c] = 1

        # if char c in t is in the dictionary, minus 1 occurence
        # if occurence = 0 or no char in dictionary at all, return false
        # if all char is found, return true
        for c in t:
            if c in table and table[c] > 0:
                table[c] = table[c] - 1
            else:
                return False
        return True

        